token-handler/src/mint_tokens.rs

```diff
@@ -1,79 +1,187 @@
-            let sov_prefix = self.sov_prefix().get();
-            if !self.has_sov_prefix(&operation_token.token_identifier, &sov_prefix) {
-                output_payments.push(operation_token.clone());
-                continue;
-            }
-
+        for operation_token in tokens.iter() {
             let mut nonce = operation_token.token_nonce;
+
             if nonce == 0 {
-                self.send().esdt_local_mint(
-                    &operation_token.token_identifier,
-                    operation_token.token_nonce,
-                    &operation_token.token_data.amount,
-                );
+                self.tx()
+                    .to(ToSelf)
+                    .typed(system_proxy::UserBuiltinProxy)
+                    .esdt_local_mint(
+                        &operation_token.token_identifier,
+                        operation_token.token_nonce,
+                        &operation_token.token_data.amount,
+                    )
+                    .sync_call();
             } else {
-                let token_data = operation_token.token_data.clone();
-                let mut arg_buffer = ManagedArgBuffer::new();
-
-                arg_buffer.push_arg(&operation_token.token_identifier);
-                arg_buffer.push_arg(token_data.amount);
-                arg_buffer.push_arg(token_data.name);
-                arg_buffer.push_arg(token_data.royalties);
-                arg_buffer.push_arg(token_data.hash);
-                arg_buffer.push_arg(token_data.attributes);
-
-                let uris = token_data.uris.clone();
-
-                if uris.is_empty() {
-                    // at least one URI is required, so we push an empty one
-                    arg_buffer.push_arg(codec::Empty);
-                } else {
-                    // The API function has the last argument as variadic,
-                    // so we top-encode each and send as separate argument
-                    for uri in &uris {
-                        arg_buffer.push_arg(uri);
-                    }
-                }
-
-                arg_buffer.push_arg(operation_token.token_nonce);
-                arg_buffer.push_arg(token_data.creator);
-
-                let output = self.send_raw().call_local_esdt_built_in_function(
-                    self.blockchain().get_gas_left(),
-                    &ManagedBuffer::from(ESDT_NFT_CREATE_FUNC_NAME),
-                    &arg_buffer,
+                let arg_buffer = self.get_nft_create_args(
+                    &operation_token.token_identifier,
+                    &operation_token.token_nonce,
+                    &operation_token.token_data,
                 );
 
-                if let Some(first_result_bytes) = output.try_get(0) {
-                    nonce = first_result_bytes.parse_as_u64().unwrap_or_default()
-                } else {
-                    nonce = 0
-                }
+                nonce = self.call_nft_create_built_in_function(&arg_buffer);
```

Comment:
I believe we don't need `nonce` variable because it will always be `operation_token.token_nonce`. We can use here the mint function directly:

```
self.send_raw().call_local_esdt_built_in_function(
    self.blockchain().get_gas_left(),
    &ManagedBuffer::from(ESDT_NFT_CREATE_FUNC_NAME),
    arg_buffer,
);
```